File: backend_cf/models/promotion.py
from sqlalchemy import Column, Integer, String, DateTime, Boolean, Text, DECIMAL
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from database import Base

class Promotion(Base):
    __tablename__ = "promotions"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(150), nullable=False)
    description = Column(Text)
    discount_percent = Column(Integer)
    discount_amount = Column(DECIMAL(10,2))
    start_date = Column(DateTime, nullable=False)
    end_date = Column(DateTime, nullable=False)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, server_default=func.now())

    # ✅ THÊM DÒNG NÀY
    promotion_products = relationship(
        "PromotionProduct",
        back_populates="promotion"
    )

File: lib/features/promotion/domain/entities/promotion.dart
class Promotion {
  final int id;
  final String name;
  final String? description;
  final int? discountPercent;
  final double? discountAmount;
  final DateTime startDate;
  final DateTime endDate;
  final List<int> productIds;
  const Promotion({
    required this.id,
    required this.name,
    this.description,
    this.discountPercent,
    this.discountAmount,
    required this.startDate,
    required this.endDate,
    required this.productIds,
  });
}